core/src/main/java/com/falmeida/tasky/core/presentation/Effect.kt

```diff
@@ -0,0 +1,25 @@
+package com.falmeida.tasky.core.presentation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collectLatest
+
+interface Effect
+@Composable
+fun <T : Effect> HandleEffects(
+    effectFlow: Flow<T>,
+    key: Any? = Unit,
+    content: suspend CoroutineScope.(effect: T) -> Unit // Use suspend for better async handling
+) {
+    LaunchedEffect(key1 = key) {
+        try {
+            effectFlow.collectLatest { effect ->
+                content(effect)  // Collect effects
+            }
+        } catch (e: Exception) {
+            e.printStackTrace() // Handle exceptions (if any)
+        }
+    }
+}
```

Comment:
Strictly speaking, you need something like this to collect one-time events 100% safely without risking to lose some:
```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```
For more info, I talk about the problem in this video: https://youtu.be/njchj9d_Lf8?si=0to4KwRHzEIECID8